### DISCUSSION TITLE
Support tsc usage

### DISCUSSION THREAD

**Discussion by Saul-Mirone** [vanilla-extract-css/vanilla-extract#280] [Ideas]
Hi vanilla team,

Thank you for your great job on this project. However, I find out that there is on proper way for my framework to use this project.

I have several packages managed in a monorepo, and I use `tsc` to compile them. Their dependency relations are managed by tsc's [project reference feature](https://www.typescriptlang.org/docs/handbook/project-references.html). If I switch to vite or esbuild, I'll have to manage it myself and I think it's a waste of the ability of typescript. BTW, here is my monorepo: https://github.com/Saul-Mirone/milkdown.

So I suggest that maybe you can add a feature, when package Foo is using `vanilla-extract`, it just compile it as typescript, and for users import this package Foo, they need to configure vanilla-extract themselves to get styles in Foo works.

**Comment by phil-lgr** [comment]
Check matt's response here: https://github.com/seek-oss/vanilla-extract/issues/231#issuecomment-889555462 I had a similar issue

**Comment by Saul-Mirone** [comment]
> 
> 
> Check matt's response here: [#231 (comment)](https://github.com/seek-oss/vanilla-extract/issues/231#issuecomment-889555462) I had a similar issue

So can I think as that it can work no matter it's js or ts?
